generators/app/files/CONTRIBUTING.md

```diff
@@ -2,9 +2,123 @@
 
 Please follow these guidelines for contributing to this repository.
 
+## Branches
+
+**Never commit directly to `master`.** Create small branches for *each* topic that you work on
+(individual features, bug fixes, and so on). When the branch is complete,
+[create a pull request](https://help.github.com/articles/using-pull-requests/) against `master`.
+
+Branches should be *small* and *short-lived*. The goal is to be able to
+merge back into `master` frequently, so that the team can access your changes,
+and so that you stay in closer sync with the trunk.
+
+Remember to *frequently* [rebase](http://git-scm.com/book/ch3-6.html) your branch,
+to get the latest changes from `master`. This will give you access to any new
+features and utilities, plus make merging upstream (much) easier.
+
+### Naming
+
+Branch names should be:
+
+- Descriptive and concise (pithy, even)
+- All lowercase
+- Separated by *underscores*
```

Comment:
Heavy personal preference for `-` instead of `_` due to a) it has nicer typographic weight (center vs baseline) and b) it's easier to type (no shift key). :bike: :house: 

Maybe others can vote on this.